### PR TITLE
Expose torch.distributed.is_available() API

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37021 Expose torch.distributed.is_available() API**

Differential Revision: [D21164318](https://our.internmc.facebook.com/intern/diff/D21164318)

### CODE REVIEW

**Comment by mrshenli** [comment]
Screen Shot 2020-04-21 at 5 20 39 PM

**Comment by mrshenli** [comment]
This is trying to address comments from this [forum post](https://discuss.pytorch.org/t/module-torch-distributed-has-no-attribute-is-initialized/55391/6)

File: torch/distributed/__init__.py [hunk 1]
```diff
@@ -4,6 +4,12 @@
 
 
 def is_available():
+    """
+    Returns ``True`` if the distributed package is available. Otherwise,
+    ``torch.distributed`` does not expose any other APIs. Currently,
+    ``torch.distributed`` is available on Linux and MacOS. Set
+    ``USE_DISTRIBUTED=1`` at compile time to enable it.
```
Comment (rohan-varma) [inline torch/distributed/__init__.py:11]: It might be helpful to specify that this is for building from source?
Also, what's the current default?

**Review by osalpekar — APPROVED** [review]

File: torch/distributed/__init__.py (hunk 1 above)
Comment (mrshenli) [reply torch/distributed/__init__.py:11]: good point! will update

current default: USE_DISTRIBUTED=1 on Linux and USE_DISTRIBUTED=0 on MacOS, let me include those

**Pull request closed without merging**